pkg/queue/bounded_queue.go

```diff
@@ -89,6 +95,25 @@ func (q *BoundedQueue) StartConsumers(num int, consumer func(item interface{}))
 	startWG.Wait()
 }
 
+// statelessConsumer wraps a consume function callback
+type statelessConsumer struct {
+	consumefn func(item interface{})
+}
+
+// Consumer consumes an item from a bounded queue
+func (c *statelessConsumer) Consume(item interface{}) {
+	c.consumefn(item)
+}
```

Comment:
I would do this more in line with stdlib/http/HandlerFunc
```suggestion
type ConsumerFunc func(item interface{})

// Consumer consumes an item from a bounded queue
func (c *ConsumerFunc) Consume(item interface{}) {
	c(item)
}
```

Reply:
That makes a lot of sense. Should this be exported?

Reply:
Should be done, marking as solved